server/SocketManager/socketmanager.js

```diff
@@ -1,11 +1,14 @@
 let sockets = {};
+let socket_user_map = {};
 
-const updateSocket = (usertoken, clientsocket) => {
+const updateSocket = (usertoken, clientsocket, user_id) => {
   sockets[usertoken] = clientsocket;
+  socket_user_map[usertoken] = user_id;
```

Comment:
The `updateSocket` function now accepts a `user_id` parameter, which is a good addition. However, ensure that the function handles cases where `usertoken` or `user_id` might be undefined or null. You might want to add validation for these parameters.
*This comment was auto-generated by **Code Bot** to provide suggestions and improvements.*